src/plugins/section_category.ts

```diff
@@ -0,0 +1,257 @@
+  item.className = `menu-item ${isDisabled ? 'disabled' : ''} ${
+    isSelected ? 'selected' : ''
+  }`
+  item.textContent = contents
+  item.addEventListener('mousedown', (event) => {
+    event.preventDefault()
+    handler(event)
+  })
+  return item
+}
+
+function createMenu(
+  view: EditorView,
+  props: SectionCategoryProps,
+  node: ManuscriptNode,
+  pos: number,
+  currCategory: string
+) {
+  const menu = document.createElement('div')
+  menu.className = 'section-category menu'
+  const existingCatsCounted = getExistingCatsCounted(view.state)
+  const categories = getSortedSectionCategories(
+    view.state,
+    node,
+    props.sectionCategories,
+    pos,
+    existingCatsCounted
+  )
+  categories.forEach((category) => {
+    const item = createMenuItem(
+      category.name,
+      () => changeNodeType(view, node, category._id, pos),
+      category.isDisabled,
+      currCategory === category._id
+    )
+    menu.appendChild(item)
+  })
+
+  document.addEventListener('mousedown', (event) => {
+    if (!menu.contains(event.target as Node)) {
+      popper.destroy()
+    }
+  })
+
+  return menu
+}
+
+function createButton(
+  view: EditorView,
+  props: SectionCategoryProps,
+  node: ManuscriptNode,
+  pos: number,
+  category: string,
+  canEdit = true
+) {
+  const arrow = document.createElement('div')
+  arrow.className = 'section-category popper-arrow'
+  const button = document.createElement('button')
+  button.innerHTML = sectionCategoryIcon
+  button.className = `section-category-button ${category && 'assigned'}`
+  if (canEdit) {
+    button.addEventListener('mousedown', (e) => {
+      e.stopPropagation()
+      popper.destroy()
+      const menu = createMenu(view, props, node, pos, category)
+      popper.show(button, menu, 'bottom-end', false, [
+        { name: 'offset', options: { offset: [0, -10] } },
+      ])
+    })
+  } else {
+    button.addEventListener('mouseenter', () => {
+      const tooltip = document.createElement('div')
+      tooltip.className = 'section-category tooltip'
+      tooltip.textContent = 'Category:'
+      const span = document.createElement('span')
+      span.textContent = getCategoryName(props.sectionCategories, category)
+      tooltip.appendChild(span)
+      tooltip.appendChild(arrow)
+      popper.show(button, tooltip, 'left', false, [
+        { name: 'offset', options: { offset: [0, 10] } },
+        { name: 'arrow', options: { element: arrow } },
+      ])
+    })
+    button.addEventListener('mouseleave', () => {
+      popper.destroy()
+    })
+  }
+
+  return button
+}
+
+function buildPluginState(
+  state: EditorState,
+  props: SectionCategoryProps
+): PluginState {
+  const decorations: Decoration[] = []
+  const can = props.getCapabilities()
+
+  state.doc.descendants((node, pos) => {
+    if (
+      node.type.name === 'section' &&
+      isEditableSectionCategory(node.attrs.category) &&
+      !isUnique(node.attrs.category)
```

Comment:
I have removed this func isEditableSectionCategory, and kept the isEditableSectionCategoryID, since it take the same param as isUnique.

Fixed for the typecasting.